app/Http/Controllers/Api/VolumeController.php

```diff
@@ -186,284 +182,32 @@ public function update(UpdateVolume $request)
      * "id": 4
      * }
      **/
-    public function clone($volumeId, $destProjectId, Request $request)
+    public function clone(CloneVolume $request)
     {
-        return DB::transaction(function () use ($volumeId, $destProjectId, $request) {
-            $project = Project::findOrFail($destProjectId);
-            $this->authorize('update', $project);
-            $volume = Volume::findOrFail($volumeId);
-            $this->authorize('update', $volume);
-            $copy = $volume->replicate();
-            $copy->name = $request->input('name', $volume->name);
-            $copy->save();
+        return DB::transaction(function () use ($request) {
+            $createSyncLimit = 10000;
 
-            if ($volume->isImageVolume()) {
-                $this->copyImages($volume, $copy, $request->input('file_ids', []));
-                $this->copyImageAnnotation($volume, $copy, $request->input('file_ids', []));
-                $this->copyImageLabels($volume, $copy, $request->input('file_ids', []));
-            } else {
-                $this->copyVideos($volume, $copy, $request->input('file_ids', []));
-                $this->copyVideoAnnotation($volume, $copy, $request->input('file_ids', []));
-                $this->copyVideoLabels($volume, $copy, $request->input('file_ids', []));
-            }
+            $project = $request->project;
+            $volume = $request->volume;
 
-            //save ifdo-file if exist
-            if ($volume->hasIfdo()) {
-                $this->copyIfdoFile($volumeId, $copy->id);
-            }
+            $nbrFiles = count($request->input('only_files', []));
 
-            $project->addVolumeId($copy->id);
+            // If too many files should be created, do this asynchronously in the
+            // background. Else the script will run in the 30 s execution timeout.
+            $job = new CloneImagesOrVideos($request);
+            if (($nbrFiles === 0 && $volume->files()->count() > 0) || $nbrFiles > $createSyncLimit) {
+                Queue::pushOn('high', $job);
+            } else {
+                Queue::connection('sync')->push($job);
+            }
```

Comment:
I think we sould always push the job on the normal (non-sync) queue. With websockets we now have a way to deal with this in the UI. This wasn't possible when the async volume creation was implemented.